validator/src/main/java/com/google/daq/mqtt/mapping/MappingAgent.java

```diff
@@ -141,25 +144,66 @@ private FamilyDiscoveryConfig getFamilyDiscoveryConfig(String family) {
     return familyDiscoveryConfig;
   }
 
-  private void reconcileDiscovery() {
+  private void mapDiscoveredDevices() {
+    List<Entry<String, Metadata>> mappedDiscoveredEntries = getMappedDiscoveredEntries();
+    Map<String, Metadata> devicesEntriesMap = getDevicesEntries();
+
+    mappedDiscoveredEntries.forEach(entry -> {
+
+      if (devicesEntriesMap.containsKey(entry.getKey())) {
+        System.err.println("Skipping existing device file for family::address = " + entry.getKey());
+        //TODO: update the existing device
+      } else {
+        // UNK denotes unknown device
+        String newDeviceName = "UNK-" + suffixToStart;
+        suffixToStart++;
+        File metadataFile = siteModel.getDeviceFile(newDeviceName, METADATA_JSON);
+        System.err.println("Writing device metadata file " + metadataFile);
+        metadataFile.getParentFile().mkdirs();
+        JsonUtil.writeFile(entry.getValue(), metadataFile);
+      }
+    });
+
+    updateProxyIdsForDiscoveryNode(mappedDiscoveredEntries);
+  }
+
+  private List<Entry<String, Metadata>> getMappedDiscoveredEntries() {
     File extrasDir = siteModel.getExtrasDir();
```

Comment:
Yah -- so I know you're just copying/extending what's already there, but I think now might be a good time to clean it up! (You know, do as I say not as I did yadda yadda...). YDYT?